### ISSUE TITLE
[Feature Request]Set random seed to each env

### ISSUE THREAD

**Issue by AndssY** [sail-sg/envpool#298]
Recently, it appears that seeds can only be sequentially assigned to environments, i.e. the seed for the i-th environment is set as i+seed. 
Is there any way to assign a completely random seed to each environment? Like input a list of random seed. Is there any solution？

**Comment by Trinkle23897** [comment]
Yes, but that requires changing the c++ code. I'll try to revive master ci this week and together implement this feature.

**Comment by Trinkle23897** [comment]
Implemented in [#349](https://github.com/sail-sg/envpool/pull/349).

With this change, `seed` can be either an `int` or a per-env sequence, so you can now do:

```python
import envpool

env = envpool.make_gym("Pong-v5", num_envs=3, seed=[3, 5, 7])
```

When `seed` is a sequence, EnvPool will use those values directly instead of forcing `seed + env_id`. The sequence length must match `num_envs`.

I also added a regression test for this behavior and ran it on a Linux dev box:

```bash
bazel test //envpool/dummy:dummy_py_envpool_test \
  --config=test \
  --test_output=all \
  --test_filter=_DummyEnvPoolTest.test_env_seed_overrides_sequential_seeding
```

Output:

```text
[ RUN      ] _DummyEnvPoolTest.test_env_seed_overrides_sequential_seeding
[       OK ] _DummyEnvPoolTest.test_env_seed_overrides_sequential_seeding
```

I also reran `make bazel-test` on `dev` after the patch, and it passed there as well.
